js/api/put/put.mjs

```diff
@@ -0,0 +1,19 @@
+import variables from "../variables/index.mjs";
+/**
+ * api call to update a specific post based on id and userinfo
+ * enterer in a form
+ * @param {*} result 
+ * @param {*} optionalEndpoint 
+ */
+export async function put (result, optionalEndpoint) {
+    try {
+        const finalURL = variables.url + variables.posts + optionalEndpoint;
+        
+        result = JSON.stringify(result);        
+        const options = variables.createOptionsBody("put", result);
+        const data = await fetch(finalURL, options);
```

Comment:
This can be a useful pattern

```suggestion
        const data = await fetch(finalURL, options);
        if (response.ok) {
          return data
        } else {
          throw new Error("PUT failed!")
        }
```